cmd/plugin/builder/README.md

```diff
@@ -35,14 +35,81 @@ Plugins will find that their `make build` command will suffice for most compile
 
 ```txt
 --artifacts string   path to output artifacts (default "artifacts")
---corepath string    path for core binary
 --ldflags string     ldflags to set on build
 --match string       match a plugin name to build, supports globbing (default "*")
 --path string        path of the plugins directory (default "./cmd/cli/plugin")
 --target string      only compile for a specific target, use 'local' to compile for host os (default "all")
 --version string     version of the root cli (required)
 ```
 
+### Build-plugins
+
+`tanzu builder plugin build` can be used to build the plugins and create artifacts that can be used with tanzu cli.
+
+The artifacts output directory structure will be different compared to what we have with `tanzu builder cli compile`
+command in a way that the `tanzu builder plugin build` command will automatically group plugins based on the given os_arch
+and `target`.
+This command will also generate a plugin manifest file (`plugin_manifest.yaml`) that can be used in the publishing and
+installation of plugins from the local directory with `tanzu plugin install --local`.
+
+Below are the flags available with this command:
+
+```txt
+      --artifacts string      path to output artifacts directory (default "./artifacts")
+  -h, --help                  help for build
+      --ldflags string        ldflags to set on build
+      --match string          match a plugin name to build, supports globbing (default "*")
+      --os-arch stringArray   compile for specific os-arch, use 'local' for host os, use '<os>_<arch>' for specific (e.g. 'linux_amd64') (default [all])
+      --path string           path of plugin directory (default "./cmd/plugin")
+  -v, --version string        version of the plugins
+```
+
+Below are a few examples:
+
+```shell
+  # Build all plugins under the 'cmd/plugin' directory for the local host os and arch
+  tanzu builder plugin build --path ./cmd/plugin --version v0.0.2 --os-arch local
+
+  # Build all plugins under the 'cmd/plugin' directory for os-arch 'darwin_amd64', 'linux_amd64', 'windows_amd64'
+  tanzu builder plugin build --path ./cmd/plugin --version v0.0.2 --os-arch darwin_amd64 --os-arch linux_amd64 --os-arch windows_amd64
+
+  # Build only foo plugin under the 'cmd/plugin' directory for all supported os-arch
+  tanzu builder plugin build --path ./cmd/plugin --version v0.0.2 --os-arch all --match foo
+```
+
+Here is the artifacts directory structure that gets generated by running the `tanzu builder plugin build` command.
```

Comment:
nit: L80-81 should match the output shown or vice-versa.
+1 to showing multiple osarch and target types to illustrate the point, so I suggest showing the full `tanzu build plugin build` command (to include the --os-arch twice)